### PR TITLE
Doc: CClient Core APIs documentation.

### LINKED ISSUE #985: CClient: Doxygen formatting on core APIs

### PR DESCRIPTION
**Pull request by oopsmonk**
Fixes #985 

# Test Plan:
No needed.

### CODE REVIEW

File: cclient/api/core/add_neighbors.h [hunk 1]
```diff
@@ -26,19 +26,20 @@ extern "C" {
 #endif
 
 /**
- * @brief
+ * @brief Add a list of neighbors to the IRI node.
+ *
+ * Assumes <b>addNeighbors</b> command is abaliable on the node. It should be noted that this is only temporary, and the
```
Comment (tsvisabo) [inline cclient/api/core/add_neighbors.h]: typo - available

File: cclient/api/core/add_neighbors.h [hunk 2]
```diff
@@ -26,19 +26,20 @@ extern "C" {
...
- * @param service client service
- * @param req Request containing list of neighbors to add
- * @param res Response containing a number of neighbor
- * @return
- *  - `#RC_OK`
- *  - `#RC_CCLIENT_OOM`
- *  - `#RC_ERROR`
+ * @param[in] service client service
+ * @param[in] req A list of URIs.
+ * @param[out] res Number of neighbors that were added.
```
Comment (tsvisabo) [inline cclient/api/core/add_neighbors.h:41]: what if result will contain more info in the future?
maybe: "res Response from the node, containing the number of neighbors that were added." ?

File: cclient/api/core/attach_to_tangle.h [hunk 3]
```diff
@@ -26,28 +26,30 @@ extern "C" {
 #endif
 
 /**
- * @brief
+ * @brief Performs Proof-of-Work required to attach a transation to the Tangle.
+ *
+ * <p>
+ * Prepares the specified transactions (trytes) for attachment to the Tangle by doing Proof of Work. You need to supply
```
Comment (tsvisabo) [inline cclient/api/core/attach_to_tangle.h:32]: Suggestion - 
"You will have to" in "You need to"

Comment (tsvisabo) [reply cclient/api/core/attach_to_tangle.h:32]: Suggestion -
"These are the tips which you're going to validate and
 * reference with this transaction." --> "The new transaction is referencing and validating these new tips"

File: cclient/api/core/broadcast_transactions.h [hunk 4]
```diff
@@ -25,17 +25,13 @@ extern "C" {
 #endif
 
 /**
- * @brief
- *
- * Broadcast a list of transactions to all neighbors. The input trytes for this
- * call are provided by iota_client_attach_to_tangle.
- *
- * https://iota.readme.io/reference#broadcasttransactions
+ * @brief Broadcast a list of transactions to all neighbors.
```
Comment (tsvisabo) [inline cclient/api/core/broadcast_transactions.h]: Suggestion - 
"to all neighbors" --> "to all node's neighbors"

File: cclient/api/core/check_consistency.h [hunk 5]
```diff
@@ -17,21 +26,15 @@ extern "C" {
...
- *
- * https://iota.readme.io/reference#checkconsistency
+ * @brief Check the consistency of the subtangle formed by the provided tails.
  *
- * @param service IRI node end point.
- * @param req - Request containing tail transaction hash (hash of
- * transaction with `currentIndex=0`), or array of tail transaction hashes.
- * @param res - Response containing consistency state of given transaction
- * or co-consistency of given transactions.
+ * As long as a transaction is consistent it might be accepted by the network. In case a transaction is inconsistent, it
```
Comment (tsvisabo) [inline cclient/api/core/check_consistency.h]: Suggestion - 
An inconsistent transaction will not be accepted by the network,  and a reattachment is required by ....

File: cclient/api/core/get_balances.h [hunk 6]
```diff
@@ -17,20 +26,14 @@ extern "C" {
...
- * list in the same order as the addresses were provided as input.
- *
- * https://iota.readme.io/reference#getbalances
+ * @brief Fetches <b>confirmed<b/> balances of given addresses at the latest solid milestone.
  *
- * @param service IRI node end point.
- * @param req Request containing the addresses for which balances are
- * requested
- * @param res Response containing the requested balances
+ * If the <b>tips</b> has not given it retruns the view by latest solid milestone.
```
Comment (tsvisabo) [inline cclient/api/core/get_balances.h]: If tips were not provided, it returns ...

File: cclient/api/core/get_transactions_to_approve.h [hunk 7]
```diff
@@ -17,20 +26,19 @@ extern "C" {
...
- * transactions).
- *
- * https://iota.readme.io/reference#gettransactionstoapprove
+ * @brief Does the <b>tip selection</b>
  *
- * @param service IRI node end point.
- * @param req Request containing number of bundles to go back to determine
- * the transactions for approval.
- * @param res Response containing transactions to approve (branch + trunk)
+ * Tip selection which returns <b>trunk</b> and <b>branch</b> transaction. The input value <b>depth</b>
```
Comment (tsvisabo) [inline cclient/api/core/get_transactions_to_approve.h:31]: Suggestion - 
Tip selection which returns two tips that can be later used as trunk and branch

File: cclient/api/core/get_transactions_to_approve.h [hunk 8]
```diff
@@ -17,20 +26,19 @@ extern "C" {
...
+ * determines how many milestones to go back for finding the transactions to approve. The higher your <b>depth</b>
+ * value, the more work you have to do as you are confirming more transactions. If the <b>depth</b> is too large
+ * (usually above 15, it depends on the node's configuration) an error will be returned. The <b>reference</b> is an
+ * optional hash of a transaction you want to approve. If it can't be found at the specified <b>depth</b> then an
```
Comment (tsvisabo) [inline cclient/api/core/get_transactions_to_approve.h:35]: you want to approve --> you want the selected tips to reference

**Review by tsvisabo — CHANGES_REQUESTED** [review]

File: cclient/api/core/add_neighbors.h (hunk 2 above)
Comment (oopsmonk) [reply cclient/api/core/add_neighbors.h:41]: Responses are all from the connected node in Core APIs, 
https://github.com/iotaledger/entangled/pull/1093/files#diff-8ef0c618acc3cbe03ab59dee66f2a621R14

File: cclient/api/core/attach_to_tangle.h (hunk 3 above)
Comment (oopsmonk) [reply cclient/api/core/attach_to_tangle.h:32]: "will have to" is a future tense, but you need to field branch/trunk in this request.  
There are no new transaction and new tips, it's this transaction and tips selected from the IRI node.

File: cclient/api/core/get_transactions_to_approve.h (hunk 8 above)
Comment (oopsmonk) [reply cclient/api/core/get_transactions_to_approve.h:35]: To approve is referenced by two tips in short. it's interchangeable in the white paper.

**Review by thibault-martinez — APPROVED** [review]

**Review by tsvisabo — APPROVED** [review]

**Pull request merged**